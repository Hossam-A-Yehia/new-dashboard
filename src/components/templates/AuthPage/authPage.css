.auth-page {
  @apply flex h-screen flex-wrap;
}

.auth-info {
  @apply w-full md:w-1/2 bg-auth-wallpaper bg-cover flex flex-col justify-center items-center relative;
}

.auth-overlay {
  @apply absolute inset-0 bg-black opacity-70 z-0;
}

.auth-content {
  @apply relative z-10 flex flex-col items-center text-center;
}

.auth-description {
  @apply text-white mt-4 font-semibold text-lg;
}

.form-container {
  @apply w-full md:w-1/2 flex flex-col justify-center items-center bg-white;
}
