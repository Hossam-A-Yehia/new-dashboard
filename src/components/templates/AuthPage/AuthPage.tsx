import React from "react";
import Logo from "@/assets/images/logo.svg";
import "./authPage.css";
import { t } from "i18next";
import CustomImage from "@/components/atoms/Image/CustomImage";

interface AuthPageProps {
  title?: string;
  FormComponent: React.FC;
}

const AuthPage: React.FC<AuthPageProps> = ({ FormComponent }) => (
  <div className="auth-page">
    <div className="relative auth-info">
      <div className="auth-overlay"></div>
      <div className="auth-content">
        <CustomImage src={Logo} alt="Logo" height={500} width={200} />
        <p className="auth-description">{t("auth.title")}</p>
      </div>
    </div>
    <div className="form-container">
      <FormComponent />
    </div>
  </div>
);

export default AuthPage;
