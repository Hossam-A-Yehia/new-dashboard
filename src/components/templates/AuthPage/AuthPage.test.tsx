import { describe, it, expect, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import AuthPage from './AuthPage';
import React from 'react';

vi.mock('@/components/atoms/Image/CustomImage', () => ({
  default: vi.fn(({ src, alt, height, width }) => (
    <img data-testid="custom-image" src={src} alt={alt} height={height} width={width} />
  )),
}));

vi.mock('i18next', () => ({
  t: vi.fn((key) => `Translated: ${key}`),
}));

vi.mock('@/assets/images/logo.svg', () => ({
  default: 'mocked-logo.svg',
}));

vi.mock('./authPage.css', () => ({}));

const MockFormComponent: React.FC = () => <div data-testid="mock-form">Mock Form</div>;

describe('AuthPage Component', () => {
  it('renders the logo image', () => {
    render(<AuthPage FormComponent={MockFormComponent} />);

    const logo = screen.getByTestId('custom-image');
    expect(logo).toBeInTheDocument();
    expect(logo).toHaveAttribute('alt', 'Logo');
    expect(logo).toHaveAttribute('src', 'mocked-logo.svg');
    expect(logo).toHaveAttribute('height', '500');
    expect(logo).toHaveAttribute('width', '200');
  });

  it('renders the FormComponent', () => {
    render(<AuthPage FormComponent={MockFormComponent} />);

    const formComponent = screen.getByTestId('mock-form');
    expect(formComponent).toBeInTheDocument();
    expect(formComponent).toHaveTextContent('Mock Form');
  });

  it('applies correct CSS classes to layout elements', () => {
    render(<AuthPage FormComponent={MockFormComponent} />);

    const authPage = screen.getByTestId('auth-page');
    expect(authPage).toHaveClass('auth-page');

    const authInfo = authPage.querySelector('.relative.auth-info');
    expect(authInfo).toHaveClass('relative auth-info');

    const authOverlay = authPage.querySelector('.auth-overlay');
    expect(authOverlay).toHaveClass('auth-overlay');

    const authContent = authPage.querySelector('.auth-content');
    expect(authContent).toHaveClass('auth-content');

    const formContainer = authPage.querySelector('.form-container');
    expect(formContainer).toHaveClass('form-container');
  });

  it('renders without title prop', () => {
    render(<AuthPage FormComponent={MockFormComponent} />);

    expect(screen.getByTestId('auth-page')).toBeInTheDocument();
    expect(screen.getByTestId('mock-form')).toBeInTheDocument();
    expect(screen.getByTestId('custom-image')).toBeInTheDocument();
  });

});

vi.mock('./AuthPage', () => ({
  default: (props: any) => (
    <div data-testid="auth-page" className="auth-page">
      <div className="relative auth-info">
        <div className="auth-overlay"></div>
        <div className="auth-content">
          <img data-testid="custom-image" src="mocked-logo.svg" alt="Logo" height={500} width={200} />
          <p className="auth-description">Translated: auth.title</p>
        </div>
      </div>
      <div className="form-container">
        <props.FormComponent />
      </div>
    </div>
  ),
}));