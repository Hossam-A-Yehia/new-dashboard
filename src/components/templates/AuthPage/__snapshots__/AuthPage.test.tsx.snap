// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthPage Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="auth-page"
    data-testid="auth-page"
  >
    <div
      class="relative auth-info"
    >
      <div
        class="auth-overlay"
      />
      <div
        class="auth-content"
      >
        <img
          alt="Logo"
          data-testid="custom-image"
          height="500"
          src="mocked-logo.svg"
          width="200"
        />
        <p
          class="auth-description"
        >
          Translated: auth.title
        </p>
      </div>
    </div>
    <div
      class="form-container"
    >
      <div
        data-testid="form-component"
      >
        Mock Form
      </div>
    </div>
  </div>
</DocumentFragment>
`;
